### PR TITLE
feat: add tree shaking support

### PR DESCRIPTION
**Pull request by ShirasawaSama**
630kb -> 397kb

### CODE REVIEW

**Comment by ShirasawaSama** [comment]
#40

**Comment by yushijinhun** [comment]
I prefer to postpone this feature until v1.2 is released, since it's not a minor change.

**Comment by ShirasawaSama** [comment]
@yushijinhun Yes, it is.

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": [["@babel/preset-env", { "modules": false }]]
```
Comment (g-plane) [inline .babelrc]: Why are you going to use Babel?

File: package.json [hunk 2]
```diff
@@ -33,9 +33,12 @@
 		"dist"
 	],
 	"dependencies": {
-		"three": "^0.112.0"
+		"@babel/core": "^7.7.7",
+		"rollup-plugin-babel": "^4.3.3",
```
Comment (g-plane) [inline package.json]: Don't add these as production dependencies.

File: .babelrc (hunk 1 above)
Comment (ShirasawaSama) [reply .babelrc]: This is my mistake. I wanted to write a Babel plugin to replace paths. I forgot to delete it.

File: package.json (hunk 2 above)
Comment (ShirasawaSama) [reply package.json]: My stupid mistake.

**Comment by ShirasawaSama** [comment]
另外是不是使用tslint去自动修复换行符, 配置里边的是 `LF`, tslint格式化之后却是 `CRLF` @g-plane

**Comment by g-plane** [comment]
TSLint 的问题先不用管，因为以后会换掉。

File: package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
 		"dist"
 	],
 	"dependencies": {
-		"three": "^0.112.0"
+		"three": "^0.111.0"
```
Comment (g-plane) [inline package.json]: 为何要降级？是有什么 breaking changes 吗？

Comment (ShirasawaSama) [reply package.json]: 好像是之前taobao源没有同步过来(三小时前), 之前安装提示没有找到这个版本, cdnjs上也没有

Comment (g-plane) [reply package.json]: 要用 npm 的官方源。

**Comment by g-plane** [comment]
建议一个 PR 只做一件事，即先不管依赖的问题，并把 `package.json` 和 `package-lock.json` 恢复。

File: src/viewer.ts [hunk 4]
```diff
@@ -24,22 +30,24 @@ export class SkinViewer {
...
-	public readonly scene: THREE.Scene;
-	public readonly camera: THREE.PerspectiveCamera;
-	public readonly renderer: THREE.WebGLRenderer;
+	public readonly scene: Scene;
+	public readonly camera: PerspectiveCamera;
+	public readonly renderer: WebGLRenderer;
 
 	public readonly playerObject: PlayerObject;
 
+	private readonly __size = new Vector2();
```
Comment (yushijinhun) [inline src/viewer.ts]: It's not wise to sacrifice readability for a little bit performance here.

**Comment by yushijinhun** [comment]
@ShirasawaSama I have rewritten the PR, please review.

**Comment by ShirasawaSama** [comment]
可以了 @yushijinhun

**Comment by yushijinhun** [comment]
I'm going to revert this PR due to https://github.com/mrdoob/three.js/issues/16059#issuecomment-479985629:
> With aliasing `three` to `src/Three.js` you'll lost both glsl and gl-constants build [optimization](https://github.com/mrdoob/three.js/blob/dev/rollup.config.js#L197-L200) and [JSM](https://github.com/mrdoob/three.js/search?q=jsm&unscoped_q=jsm&type=Issues) imports support that fixes this long standing [issue](https://github.com/mrdoob/three.js/issues/9562).

It's a upstream problem, so skinview3d can't help solve it.

**Comment by ShirasawaSama** [comment]
具体会有什么影响呢

另外其实只有 `.all.js` 这个文件受到了影响, 本来的非 `.all.js` 打包的文件是不受影响的.

因为在 `rollup.config.js` 重写了导入的模块, 把 `three/src/*` 全部指回了 `three`

**Comment by yushijinhun** [comment]
Bundling dependencies should only be done by end users, so I'm not going to include `*.all.js` in skinview3d distribution.

There are many discussions on bundling/tree-shaking three.js. I need to do some further research to find a better solution. This PR is actually a bad solution.

Further discussion goes to #40.

**Pull request merged**